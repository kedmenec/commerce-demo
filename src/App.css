html {
  display: flex;
  height: 100%;
}

body, #root {
  font-family: 'Roboto', sans-serif;
  flex: 1;
  display: flex;
  flex-direction: column;
}
