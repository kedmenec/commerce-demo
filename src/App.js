import React, { Component } from 'react';
import './App.css';


import {
  Link
} from 'react-router-dom'

class App extends Component {
  render() {
    return (
      <div style={{display: 'flex',flexDirection: 'row', flex: 1}}>
        <div style={{flex:1, background: '#f7f7f7'}}/>

        <div style={{width:1200, display: 'flex', flexDirection: 'column', background: '#f7f7f7', margin: 10}}>

          <div style={{background: 'gray', border: '1px solid black', display: 'flex', flexDirection: 'row', padding: 10}}>
            <div style={{flex: 1}}>PAG Commerce</div>
            <div style={{flex: 5, display: 'flex'}}>
              <input style={{flex: 1}}/>
            </div>
            <div style={{flex: 1}}>
              <Link to='/cart'>Cart</Link>
            </div>
          </div>

          <div style={{height: 10}}></div>
          <div style={{flex: 1, border: '1px solid black', background: 'white'}}>
            {this.props.children}
          </div>

        </div>

        <div style={{flex:1, background: '#f7f7f7'}}/>
      </div>

    );
  }
}

export default App;
