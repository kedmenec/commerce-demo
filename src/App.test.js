// An example of testing a component that uses redux and redux-form

import React from 'react';
import ReactDOM from 'react-dom';
import {App} from './App';
import {MemoryRouter} from 'react-router-dom';
import {reduxForm} from 'redux-form';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';

const Decorated = reduxForm({form: 'testForm'})(App);
const store = createStore(() => ({
  products: {
    products: [
      {
        title: 'testing'
      }
    ]
  },
  navigation: {
    filter: 'Sport'
  }
}));

it('renders correctly', () => {
  const tree = renderer
    .create(
    <Provider store={store}>
      <MemoryRouter>
        <Decorated/>
      </MemoryRouter>
    </Provider>
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
