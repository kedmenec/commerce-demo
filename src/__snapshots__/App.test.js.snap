exports[`test renders correctly 1`] = `
<div
  style={
    Object {
      "background": "#f7f7f7",
      "display": "flex",
      "flex": 1,
      "flexDirection": "row",
    }
  }>
  <div
    style={
      Object {
        "flex": 1,
      }
    } />
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
        "margin": 10,
        "width": 1200,
      }
    }>
    <div
      className="ui fluid card"
      href={undefined}
      onClick={[Function]}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 0,
          "padding": 10,
        }
      }>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
          }
        }>
        <a
          href="/products"
          onClick={[Function]}>
          <i
            aria-hidden="true"
            className="green new pied piper big icon"
            style={
              Object {
                "marginBottom": 10,
              }
            } />
          <b>
            CommercePiper
          </b>
        </a>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flex": 5,
          }
        }>
        <div
          className="ui input"
          style={
            Object {
              "flex": 1,
            }
          }>
          <input
            name="search"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            type="text"
            value="" />
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }>
        <a
          href="/cart"
          onClick={[Function]}>
          <div
            className="ui labeled button"
            onClick={[Function]}>
            <button
              className="ui blue basic button"
              tabIndex={0}>
               
              Cart
            </button>
            <div
              className="ui blue left pointing basic label"
              onClick={[Function]}>
              <i
                aria-hidden="true"
                className="cart icon" />
            </div>
          </div>
        </a>
      </div>
    </div>
    <div
      className="ui fluid card"
      href={undefined}
      onClick={[Function]}
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginBottom": 0,
          "padding": 10,
        }
      }>
      <div
        style={
          Object {
            "display": "flex",
            "flex": 6,
          }
        }>
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
            }
          }>
          <div
            className="ui menu"
            style={
              Object {
                "flex": 1,
              }
            }>
            <a
              className="item"
              onClick={[Function]}>
              
            </a>
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }>
        <button
          className="ui blue basic button"
          onClick={[Function]}
          tabIndex={undefined}>
          Login/Signup
        </button>
      </div>
    </div>
    <div
      className="ui fluid card"
      href={undefined}
      onClick={[Function]}
      style={
        Object {
          "flex": 1,
        }
      } />
  </div>
  <div
    style={
      Object {
        "background": "#f7f7f7",
        "flex": 1,
      }
    } />
</div>
`;
